Scripts/download-portal-item-data.swift

```diff
@@ -200,16 +237,38 @@ portalItems.forEach { (portalURLString, portalItems) in
     portalItems.forEach { (portalItem) in
         // Have we already downloaded the item?
         let filename = downloadedItems[portalItem.identifier] ?? portalItem.filename
-        if FileManager.default.fileExists(atPath: destinationURLProvider.makeURL(filename: filename).path) {
+        
+        // Check if it is a non-archive single file.
+        let isFileExist: Bool = FileManager.default.fileExists(atPath: destinationURLProvider.makeURL(filename: filename).path)
+        
+        // Check if there is a sub-folder for the corresponding archive, and the sub-folder is empty or not.
+        // The corresponding sub-folder has the same name as the archive without the extension.
+        let subFolderURL = destinationURLProvider.makeSubFolderURLForArchive(folderName: (filename as NSString).deletingPathExtension)
+        let paths = try? FileManager.default.contentsOfDirectory(
+            at: subFolderURL,
+            includingPropertiesForKeys: nil,
+            options: .skipsHiddenFiles
+        )
+        /// true: the folder exists and is empty;
+        /// false: the folder exists and has content;
+        /// nil: the folder does not exist or not accessible
+        let isEmptySubFolder = paths?.isEmpty
+        
+        if isEmptySubFolder != nil && !isEmptySubFolder! {
+            // One exception to this case is that the content in the folder is changed. There is no way to detect that
+            // unless we keep track of each extracted file in the .downloaded_items dictionary.
+            print("Item \(portalItem.identifier) has already been downloaded, and is extracted to an folder")
+            downloadedItems[portalItem.identifier] = filename
+        } else if isFileExist && isEmptySubFolder == nil {
             print("Item \(portalItem.identifier) has already been downloaded")
-            // This is a temporary measure for users who currently don't have a
-            // downloaded items file.
+            // This is a temporary measure for users who currently don't have a downloaded items file.
```

Comment:
I forgot to mention that the line break here was intentional to keep the comment to a readable width, using the default page guide at column 80:

![image](https://user-images.githubusercontent.com/2257493/82228894-93ea7800-98de-11ea-9a1f-f6b0a454a9e6.png)


  |  
--- | ---
Before | ![image](https://user-images.githubusercontent.com/2257493/82228579-22aac500-98de-11ea-8b3f-f7715915ab39.png)
After | ![image](https://user-images.githubusercontent.com/2257493/82228612-335b3b00-98de-11ea-8527-684c0d69e78f.png)

I try to wrap all of my comments to the page guide, but especially documentation comments.

Reply:
Oh thanks for pointing that out! I once asked someone the line wrapping width but it wasn't clear so I kept around 120... I'll change it to 80!